packages/cadl-ranch-specs/http/lro/lro-rpc/main.tsp

```diff
@@ -0,0 +1,202 @@
+@scenarioService("/azure/lro/rpc")
+@versioned(Azure.Lro.Rpc.Versions)
+namespace Azure.Lro.Rpc;
+
+enum Versions {
+  @useDependency(global.Azure.Core.Versions.v1_0_Preview_2)
+  v2022_12_01_preview: "2022-12-01-preview",
+}
+
+interface ResourceOperations
+  extends global.Azure.Core.ResourceOperations<NoConditionalRequests & NoRepeatableRequests & NoClientRequestId> {}
+
+@doc("Data of the job")
+model JobData {
+  @doc("Comment.")
+  comment: string;
+}
+
+@resource("jobs")
+@doc("Result of the job")
+model JobResult {
+  @doc("A processing job identifier.")
+  @visibility("read")
+  @key
+  jobId: string;
+
+  @doc("Comment.")
+  @visibility("read")
+  comment: string;
+
+  @doc("The status of the processing job.")
+  @visibility("read")
+  status: Foundations.OperationState;
+
+  @doc("Error objects that describes the error when status is \"Failed\".")
+  @visibility("read")
+  errors?: Foundations.ErrorResponse[];
+
+  @doc("The results.")
+  @visibility("read")
+  results: string[];
+}
+
+@resource("jobs/operations")
+@doc("Result of the poll")
+model JobPollResult {
+  @doc("Operation identifier.")
+  @visibility("read")
+  @key
+  operationId: string;
+
+  @doc("The status of the processing job.")
+  @visibility("read")
+  status: Foundations.OperationState;
+}
+
+@scenario
+@doc("Creates a Job")
+@summary("Creates a Job")
+@scenarioDoc("""
+Expected verb: POST
+Expected request body:
+```json
+{
+  "comment": "async job"
+}
+```
+
+Expected status code: 202
+Expected response header: operation-location={endpoint}/same-poll-result/jobs/job1
+Expected response body:
+```json
+{
+  "jobId": "job1",
+  "comment": "async job",
+  "status": "InProgress"
+}
+```
+
+Expected verb: GET
+Expected URL: {endpoint}/same-poll-result/jobs/job1
+
+Expected status code: 200
+Expected response body:
+```json
+{
+  "jobId": "job1",
+  "comment": "async job",
+  "status": "InProgress"
+}
+```
+
+Expected verb: GET
+Expected URL: {endpoint}/same-poll-result/jobs/job1
+
+Expected status code: 200
+Expected response body:
+```json
+{
+  "jobId": "job1",
+  "comment": "async job",
+  "status": "Succeeded",
+  "results": ["job1 result"]
+}
+```
+""")
+interface SamePollResult {
+  @doc("Creates a Job")
+  @route("/same-poll-result/jobs")
+  @pollingOperation(SamePollResult.getJob)
+  createJob is LocalCore.LongRunningRpcOperation<JobData, JobResult>;
+
+  @doc("Gets the status of a Job")
+  @route("/same-poll-result")
+  getJob is ResourceOperations.ResourceRead<JobResult>;
+}
+
+@scenario
+@doc("Creates a Job")
+@summary("Creates a Job")
+@scenarioDoc("""
+Expected verb: POST
+Expected request body:
+```json
+{
+  "comment": "async job"
+}
+```
+
+Expected status code: 202
+Expected response header: operation-location={endpoint}/different-poll-result/jobs/operations/operation1
+Expected response header: location={endpoint}/different-poll-result/jobs/job1
+Expected response body:
+```json
+{
+  "operationId": "operation1",
+  "status": "InProgress"
+}
+```
```

Comment:
Added a case of polling location/result is different from final resource location/result.

`OperationResourcePollingStrategy` in Java supports this legacy scenario. However it is possible not all SDK supports it.

Also the result of `getLroMetadata` API call would confuse on the `logicalResult` or `finalStep`.

Hence it likely need adjustment, after the LongRunningRpcOperation been put to Azure.Core.